web-ng/src/app/services/data-store/data-store.service.ts

```diff
@@ -73,6 +73,27 @@ export class DataStoreService {
       .set({ title: { en: newTitle } }, { merge: true })
       .then(() => projectId);
   }
+
+  // TODO: Define return types for methods in this class
+  updateProjectLayer(projectId: string, layerId: string, layer: Layer) {
+    const updatedLayer = layer as { id: string };
```

Comment:
Since in TypeScript objects are passed by reference, this will modified to provided Layer instead of making a copy. With the new ES7 spread operator you can do some magic:
```
  const { id: layerId, ...layerDoc } = layer;
```

That will give you two variables, one with the id, and `layerDoc` with everything else.

You can also remove `layerId` from the function params.